docs/0002-N-tuple-tree.md

```diff
@@ -0,0 +1,180 @@
+
+### short object root
+
+Once a Storage Hierarchy has been created, the name of each OCFL Object Root directory should also be determined from the identifier in a consistent manner. The default approach is to use the full (stripped) identifier but, if identifiers are long or there is the need to keep Storage Hierarchy paths short because of object complexity, there is the option to just use the portion of the identifier that remains after Storage Hierarchy path generation. To continue the UUID example the full OCFL Object Root path could be:    
+* **short object root = FALSE** /f81/d4f/ae7/f81d4fae7dec11d0a76500a0c91e6bf6/
+* **short object root = TRUE** /f81/d4f/ae7/dec11d0a76500a0c91e6bf6/
+
+## Examples
+
+These examples are taken from the OCFL Implementation notes:
+
+* *Flat*: Each object is contained in a directory with a name that is simply derived from the unique identifier of the object.
+  * identifer length = 12
+  * case mapping = ToLower
+  * invert mapping = FALSE
+  * tuple size = 12
+  * number of tuples = 0
+  * short object root = FALSE
+
+                [storage_root]
+                    ├── 0=ocfl_1.0
+                    ├── ocfl_1.0.html (optional copy of the OCFL specification)
+                    ├── d45be626e024
+                    |   ├── 0=ocfl_object_1.0
+                    |   ├── inventory.json
+                    |   ├── inventory.json.sha512
+                    |   └── v1...
+                    ├── d45be626e036
+                    |   ├── 0=ocfl_object_1.0
+                    |   ├── inventory.json
+                    |   ├── inventory.json.sha512
+                    |   └── v1...
+                    ├── 3104edf0363a
+                    |   ├── 0=ocfl_object_1.0
+                    |   ├── inventory.json
+                    |   ├── inventory.json.sha512
+                    |   └── v1...
+                    └── ...
+                
+
+* PairTree: [PairTree] is designed to overcome the limitations on the number of files in a directory that most file systems have. It creates hierarchy of directories by mapping identifier strings to directory paths two characters at a time. For numerical identifiers specified in hexadecimal this means that there are a maximum of 256 items in any directory which is well within the capacity of any modern filesystem. However, for long identifiers, pairtree creates a large number of directories which will be sparsely populated unless the number of objects is very large. Traversing all these directories during validation or rebuilding operations can be slow.
```

Comment:
A key part of the pairtree spec, that is not addressed anywhere in this extension, is [identifier string cleaning](https://tools.ietf.org/html/draft-kunze-pairtree-01#section-3). Again, I understand not wanting to get too far off into the weeds in this extension, but, again, I think it's important for the extension to actually be generally usable. Somewhere, in this extension or elsewhere, the issue of dealing with problematic characters in object ids must be addressed. As it stands, this extension is unusable on certain systems if your ids include common characters like `/`, `.`, and `:`.

Reply:
Good point - there definitely needs to be some wording to that effect here but the detail of the whole thing is probably another extension. Possibly a list of identifier names, formats and RegEx to map to a clean string.

Reply:
Created issue #5